plugins/redis/redis.cpp

```diff
@@ -1062,50 +1114,126 @@ unsigned __int64 Connection::publish(ICodeContext * ctx, const char * keyOrChann
 //--------------------------------------------------------------------------------
 ECL_REDIS_API unsigned __int64 ECL_REDIS_CALL SyncRPub(ICodeContext * ctx, const char * keyOrChannel, size32_t messageSize, const char * message, const char * options, int database, const char * password, unsigned timeout, bool lockedKey, bool cachedConnectionRequested)
 {
-    Owned<Connection> master = Connection::createConnection(ctx, cachedPubConnection, options, DUMMY_IP, DUMMY_PORT, true, 0, password, timeout, cachedConnectionRequested);
-    return master->publish(ctx, keyOrChannel, messageSize, message, database, lockedKey);
+    ConnectionContainer master;
+    try
+    {
+        master.setown(Connection::createConnection(ctx, cachedPubConnection, options, DUMMY_IP, DUMMY_PORT, true, 0, password, timeout, cachedConnectionRequested));
+        return master->publish(ctx, keyOrChannel, messageSize, message, database, lockedKey);
+    }
+    catch (IException * error)
+    {
+        master.handleException(error);
+    }
+    throwUnexpected();
```

Comment:
I'm not sure if this is here to avoid warnings about "no return" - if so it might be better and clearer to mark handleException as noreturn.

Alternatively, since master->handleException doesn't really handle the exception at all, I would have structured this as 

```
catch (...)
{
   master.freeContext();
   throw;
}
```